qiita_db/metadata_template/constants.py

```diff
@@ -6,10 +6,68 @@
 # The full license is in the file LICENSE, distributed with this software.
 # -----------------------------------------------------------------------------
 
+from collections import namedtuple
+from future.utils import viewkeys, viewvalues
+
+Restriction = namedtuple('Restriction', ['columns', 'error_msg'])
+
+# A dict containing the restrictions that apply to the sample templates
+SAMPLE_TEMPLATE_COLUMNS = {
+    'EBI': Restriction(columns={'collection_timestamp': 'timestamp',
+                                'physical_specimen_location': 'varchar'},
+                       error_msg="EBI submission disabled"),
+    'Qiita_main': Restriction(columns={'sample_type': 'varchar',
```

Comment:
Got it, yeah it makes more sense that way. Otherwise it is one of those
seemingly arbitrary restrictions of the system. If these are only for us
then they should be parametrized, although we do want to provide Qiita
as a service, the software should not be specific to our needs in any
way.

On (Apr-27-15|15:12), josenavas wrote:
> > @@ -6,10 +6,68 @@
> >  # The full license is in the file LICENSE, distributed with this software.
> >  # -----------------------------------------------------------------------------
> >  
> > +from collections import namedtuple
> > +from future.utils import viewkeys, viewvalues
> > +
> > +Restriction = namedtuple('Restriction', ['columns', 'error_msg'])
> > +
> > +# A dict containing the restrictions that apply to the sample templates
> > +SAMPLE_TEMPLATE_COLUMNS = {
> > +    'EBI': Restriction(columns={'collection_timestamp': 'timestamp',
> > +                                'physical_specimen_location': 'varchar'},
> > +                       error_msg="EBI submission disabled"),
> > +    'Qiita_main': Restriction(columns={'sample_type': 'varchar',
> 
> Re documentation: I will add a comment here and I've created a new issue: https://github.com/biocore/qiita/issues/1111
> 
> I think I used the wrong words on my previous comment, the idea is that those restrictions only apply to our system. Other labs that may want to have a qiita installation locally with multiple users shouldn't have these restrictions either... Is one of those cases in which our system is more specialized than the rest of the users.
> 
> Thinking a bit more about this. I think that having this in a configuration file will allow us to change these restrictions dynamically, and we can have them documented there. What do you think?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/qiita/pull/1075/files#r29197135